Sources/RInAppMessaging/Utilities/ResponseStateMachine.swift

```diff
@@ -0,0 +1,23 @@
+import Foundation
+
+internal struct ResponseStateMachine {
+    internal enum State {
+        case success
+        case error(Error)
+    }
+    private(set) var previousState = State.success
+    private(set) var state = State.success
+    private(set) var consecutiveErrorCount = 0
```

Comment:
Maybe we can declare these variables as private(set) internal?

Reply:
That's already how it is now. internal is implicit. I'm putting internal in class/enum/struct declarations to make it clear that it's not public. For properties I don't explicitly write internal. Do you think it's better to use explicit declaration?

Reply:
Ok. I thought it was mandatory to use `internal` in the IAM project.

Reply:
I see. I don't remember the conversation but I think the requirement was just for class/enum/struct/protocol declarations only.